### PR TITLE
Add the Anthropic provider to Gateway

### PR DESCRIPTION
**Pull request by BenWilson2**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Add anthropic as a provider

## How is this patch tested?

- [ ] Existing unit/integration tests
- [X] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [X] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/gateway/providers/anthropic.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        async with aiohttp.ClientSession(headers=headers) as session:
+            url = "/".join([config["anthropic_api_base"].rstrip("/"), path.lstrip("/")])
+            async with session.post(url, json=payload) as response:
+                js = await response.json()
+                try:
+                    response.raise_for_status()
+                except aiohttp.ClientResponseError as e:
+                    detail = js.get("error", {}).get("message", e.message)
+                    raise HTTPException(status_code=e.status, detail=detail)
+                return js
```
Comment (harupy) [inline mlflow/gateway/providers/anthropic.py:29]: We might want to put this in a util function, out of scope of this PR.

File: mlflow/gateway/providers/anthropic.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        # {
+        #     "completion": " Here is a basic overview of how a car works:\n\n1. The engine. "
+        #                   "The engine is the power source that makes the car move."
+        #     "stop_reason": "max_tokens",
+        #     "model": "claude-instant-1.1",
+        #     "truncated": False,
+        #     "stop": None,
+        #     "log_id": "dee173f87ddf1357da639dee3c38d833",
+        #     "exception": None,
+        # }
```
Comment (harupy) [inline mlflow/gateway/providers/anthropic.py:78]: Does Anthropic have an API reference like the one OpenAI has?

File: mlflow/gateway/providers/anthropic.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+from .base import BaseProvider
+from ..schemas import completions, chat, embeddings
+
+
+class AnthropicProvider(BaseProvider):
+    NAME = "anthropic"
+    SUPPORTED_ROUTES = "completions"
+
+    async def _request(self, path: str, payload: Dict[str, Any]):
+        import aiohttp
```
Comment (harupy) [inline mlflow/gateway/providers/anthropic.py:15]: Should we include `atiohttp` as a gateway dependency?

File: mlflow/gateway/providers/anthropic.py (hunk 2 above)
Comment (BenWilson2) [reply mlflow/gateway/providers/anthropic.py:78]: it does, but not as fully-featured. The examples do not match the actual response fields that are returned when querying their REST APIs. I updated with a link to their docs and matching the example to the fields that are in the docs.

File: mlflow/gateway/providers/anthropic.py (hunk 3 above)
Comment (BenWilson2) [reply mlflow/gateway/providers/anthropic.py:15]: yes, good idea. I'll make that change.

**Review by harupy — APPROVED** [review]

**Pull request merged**